src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -537,8 +537,20 @@ event::ptr primitive_inst::realloc_if_needed() {
         }
     }
 
+    bool has_loop_user = false;
+    for (const auto& user : _node->get_users()) {
+        if (user->is_type<loop>()) {
+            has_loop_user = true;
+        }
+    }
+
     // Clear out memory if if was previously reused, but now primitive can't be optimized
-    if (_node->is_type<gather>() || _node->is_type<permute>() || _node->is_type<reshape>() || _node->is_type<reorder>() || _node->is_type<strided_slice>()) {
+    if (!has_loop_user &&
+        (_node->is_type<gather>() ||
+        _node->is_type<permute>() ||
+        _node->is_type<reshape>() ||
+        _node->is_type<reorder>() ||
+        _node->is_type<strided_slice>())) {
```

Comment:
Is your case for optimized <gather|permute|reorder|strided_slice> + loop ? 
In that case, what does it mean ignoring the below codes?

Reply:
In case input node of loop is reshape(or gather/permute/reorder/strided_slice) and 2nd inference has smaller output memory of the input node than 1st infer, the output mem is not updated at this time. This causes not-matched shape issue at the inner network of the loop.

Reply:
Yes, this was the WA code for the case.
I will find the way for loop to use input's primitive_impl shape instead of input "memory" shape as you pointed.

Reply:
Deleted WA code and updated by reinterpreting with impl_shape.